Gemfile

```diff
@@ -2,8 +2,8 @@ source("https://rubygems.org")
 
 gem "xcode-install", ">= 2.6.7" # needed for running xcode-install related tests
 
-gem "danger", "~> 8.0"
-gem "danger-junit", "~> 1.0"
+gem "danger", "~> 8"
+gem "danger-junit", "~> 1"
```

Comment:
Hi @AliSoftware 👋 thanks for your review! 😁 

>Any particular reason for this change?

Objectively: no, leftover changes from unsuccessful attempts to do something else 😅 

After reading your comment, though, I don't think what you described is what actually happens 🤔 ? My understanding from what you wrote is that if I set danger to be `~> 8`, it would bump to v9 (assuming it exists), and it doesn't. Here's my test:

1. I picked a random repo I have with a Gemfile and set `gem "rake", "~> 12"` (it was `gem "rake"` before, with no version specification)
2. `bundle install`
3. Rake got downgraded from 13 to 12:

<img width="733" alt="image" src="https://github.com/fastlane/fastlane/assets/8419048/82c2905f-31b8-471d-9feb-97aa6c5affe5">

4. `bundle update rake`
5. Nothing changes:

<img width="732" alt="image" src="https://github.com/fastlane/fastlane/assets/8419048/92e3d0c4-65f6-48e3-854b-dc0cc440cf36">

6. Update to `gem "rake", "~> 13"`
7. `bundle install`
8. Gem gets updated to v13:

<img width="734" alt="image" src="https://github.com/fastlane/fastlane/assets/8419048/907ecf6c-7c5d-4327-9bee-71bc98cfa6ec">

Am I missing something here? 😮 

Happy to revert to the previous syntax btw, but I legitimately want to understand the right behavior 😅

Reply:
Agreed, I think this change is fine even if not strictly needed to fix CI

Reply:
Ah I'm guessing that TIL that there's an exception about the behavior of `~>` when only one component is provided then!

ie it's probably something like "more than one component = ok to bump the last provided version component but not the previous ones" but "only one component = not ok to bump the last (and only) provided version component" (in other words `~> 8` is equivalent to `~> 8.0`)

Reply:
Ah, confirmed by [this RubyGems Guide](https://guides.rubygems.org/patterns/#pessimistic-version-constraint):

> It’s also important to know that if you specify a major version only, like this:
> ```
> # gemspec
> spec.add_runtime_dependency 'library', '~> 2'
> ```
> It will only use the latest version from the 2.x series – so 2.3.0 – and not 3.0.0. This behaviour may surprise some people, but automatically allowing any major version past version 2 is more surprising behaviour.

Seems I was part of those "some people" to be surprised by this 😄

Reply:
Nice find!! 🥳 
Thanks for confirming 😉
Can we leave it like this then? :)

Reply:
Tbh I'd personally revert back to the old syntax (1) because it's less surprising (no need for the reader to know about that particular exception about the `~>` operator) and (2) because this change is not needed to fix the CI (so changing the line nonetheless less might create noise on future potential `git blame` and `git bisect`

But ultimately it's up to you as both are correct 🙃

Reply:
🤝 I'll revert

Reply:
Ended up splitting in 2 commits 🤦 but it's there 😊